flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/LocalExecutorITCase.java

```diff
@@ -649,6 +703,57 @@ public void testBatchQueryExecution() throws Exception {
 		}
 	}
 
+	@Test(timeout = 90_000L)
+	public void testBatchQueryCancel() throws Exception {
+		final Map<String, String> replaceVars = new HashMap<>();
+		replaceVars.put("$VAR_PLANNER", planner);
+		replaceVars.put("$VAR_EXECUTION_TYPE", "batch");
+		replaceVars.put("$VAR_RESULT_MODE", "table");
+		replaceVars.put("$VAR_MAX_ROWS", "100");
+		final LocalExecutor executor = createModifiedExecutor(
+				DEFAULTS_SIMPLE_ENVIRONMENT_FILE, clusterClient, replaceVars);
+		final SessionContext session = new SessionContext("test-session", new Environment());
+		String sessionId = executor.openSession(session);
+		assertEquals("test-session", sessionId);
+
+		try {
+			// create a bounded source, which will send a record per second until 100000000 records.
+			String ddl = "CREATE TABLE MyRandomSource (\n" +
+					"  a INT,\n" +
+					"  b BIGINT\n" +
+					") WITH (\n" +
+					"  'connector.type' = 'random',\n" +
+					"  'random.limit' = '100000000',\n" +
```

Comment:
Can't we make the limit +Infinity (or Long.MAX_VALUE) and turn this "bounded" source into an infinite source?

Reply:
we can limit that